lab/ctl/rpc.py

```diff
@@ -76,6 +79,74 @@ def get_pymod(self, modname, venvpath=None):
         return self.get_pymods([modname], venvpath=venvpath)[modname]
 
 
+class Execnet(object):
+    def __init__(self, config, location, **kwargs):
+        self.config = config
+        self.location = location
+        self.gw = self.from_location(location)
+        self._modsrc = {}
+
+    def makespec(self, location):
+        # build gw spec
+        spec = {}
+        spec['id'] = "@".join((self.__class__.__name__, location.hostname))
+        # ssh spec
+        facts = location.facts
+        user = facts.get('user', facts.get('login', 'root'))
+        sshspec = "{user}@{hostname}".format(user=user,
+                                             hostname=location.hostname)
+        keyfile = facts.get('keyfile')
+        if keyfile:
+            sshspec = "-i {} ".format(keyfile) + sshspec
```

Comment:
oddly enough changing this causes me to be prompted to unlock my key?

```bash
Warning: Identity file 'id_rsa.pytest' not accessible: No such file or directory.                    
Warning: Permanently added 'sip-cannon.qa.sangoma.local,10.10.8.19' (RSA) to the list of known hosts.
Enter passphrase for key '/home/tyler/.ssh/id_rsa':
```

So i'm just going to leave as is for now and we can investigate down the road if someone wants.